### ISSUE TITLE
TTK Persistance diagram clustering: bottleneck distance from ParaView

### ISSUE THREAD

**Issue by CharlesGueunet** [topology-tool-kit/ttk#565]
Hello,

If my understanding is correct, we should be able to use the Bottleneck distance for the TTK Persistance Diagram Clustering filter by setting the p parameter to "inf".
It is unfortunately not possible to type Inf in the dialog inside ParaView. We may want to change this mechanism.

Best,
Charles

**Comment by julien-tierny** [comment]
hi charles,

thanks for your feedback.

> If my understanding is correct, we should be able to use the Bottleneck distance for the TTK Persistance Diagram Clustering filter by setting the p parameter to "inf".
actually, not quite.

in principle, the wasserstein distance becomes equivalent to the bottleneck distance indeed when p goes to infinity. however in practice, two distinct algorithms need to be used for these two distances (taking very high values of p, e.g. 10e23, makes the wasserstein computation numerically meaningless).

the module bottleneck distance actually implements two distinct algorithms, one for the exact computation of the wasserstein distance, one for the exact computation of the bottleneck distance (triggered by entering "inf").

when computing distances between two diagrams, the module "PersistenceDiagramClustering" actually uses a fast and reliable approximation of the wasserstein distance.
note also that the computation of barycenters and clusters is only well defined for p=2 (the p parameters is only taken into account for distances).

so in short, this is how it's meant to be :)

cheers,
-- 
Dr Julien Tierny
CNRS Researcher
Sorbonne Universite
http://lip6.fr/Julien.Tierny
